giraffe/src/transforms/annotation.ts

```diff
@@ -0,0 +1,68 @@
+import {AnnotationLayerSpec, AnnotationMark, Table} from '../types'
+import {FILL} from '../constants/columnKeys'
+import {createGroupIDColumn} from './'
+
+export const annotationTransform = (
+  inputTable: Table,
+  inputAnnotations: AnnotationMark[],
+  xColumnKey: string,
+  yColumnKey: string,
+  fillColKeys: string[]
+): AnnotationLayerSpec => {
+  const [fillColumn] = createGroupIDColumn(inputTable, fillColKeys)
+  const table = inputTable.addColumn(FILL, 'system', 'number', fillColumn)
+  const xCol = table.getColumn(xColumnKey, 'number')
+  const yCol = table.getColumn(yColumnKey, 'number')
+
+  let xMin = Infinity
+  let xMax = -Infinity
+  let yMin = Infinity
+  let yMax = -Infinity
```

Comment:
why are the max values negative infinity?

Reply:
Those are initial values. By setting a max with a very small start value (negative Infinity), we guarantee that the first value will become the next max. Same concept for min and using a very large start value.